test/unit/app/utils/test_kenya_state_machine.py

```diff
@@ -0,0 +1,137 @@
+current_pin_state = partial(UssdSessionFactory, state="current_pin")
+new_pin_state = partial(UssdSessionFactory, state="new_pin")
+new_pin_confirmation_state = partial(UssdSessionFactory, state="new_pin_confirmation", user_input="2222*2222")
+opt_out_of_market_place_pin_authorization_state = partial(UssdSessionFactory,
+                                                          state="opt_out_of_market_place_pin_authorization")
+exchange_token_state = partial(UssdSessionFactory, state="exchange_token")
+exchange_rate_pin_authorization_state = partial(UssdSessionFactory, state="exchange_rate_pin_authorization")
+exchange_token_agent_number_entry_transitions_state = partial(UssdSessionFactory, state="exchange_token_agent_number_entry_transitions")
+exchange_token_amount_entry_state = partial(UssdSessionFactory, state="exchange_token_amount_entry")
+exchange_token_pin_authorization_state = partial(UssdSessionFactory, state="exchange_token_pin_authorization")
+exchange_token_confirmation_state = partial(UssdSessionFactory, state="exchange_token_confirmation")
+
+
+@pytest.mark.parametrize("session_factory, user_factory, user_input, expected",
+                         [
+                             # initial_language_selection transitions tests
+                             (initial_language_selection_state, standard_user, "1", "complete"),
+                             (initial_language_selection_state, standard_user, "2", "complete"),
+                             (initial_language_selection_state, standard_user, "3", "help"),
+                             (initial_language_selection_state, standard_user, "5", "exit_invalid_menu_option"),
+                             # initial_pin_entry transitions tests
+                             (initial_pin_entry_state, standard_user, "0000", "initial_pin_confirmation"),
+                             (initial_pin_entry_state, standard_user, "AAAA", "exit_invalid_pin"),
+                             # initial_pin_confirmation transitions tests
+                             (initial_pin_confirmation_state, standard_user, "0000", "complete"),
+                             (initial_pin_confirmation_state, standard_user, "1212", "exit_pin_mismatch"),
+                             # start state transition tests
+                             (start_state, standard_user, "1", "send_enter_recipient"),
+                             (start_state, standard_user, "2", "account_management"),
+                             (start_state, standard_user, "3", "directory_listing"),
+                             (start_state, standard_user, "4", "exchange_token"),
+                             (start_state, standard_user, "5", "help"),
+                             (start_state, standard_user, "6", "exit_invalid_menu_option"),
+                             # send_enter_recipient state tests
+                             # (send_enter_recipient_state, standard_user, "0712345678", "send_token_amount"),
```

Comment:
i'd put TODOs around these so it's more clear which ones are comments vs things that need to be fixed later

Reply:
I'll add the TODOs on top of these and push